### PR TITLE
Chore: remove legacy components and CSS classes for OpenTsdbDetails Component

### LINKED ISSUE #76107: Chore [Epic]: Replace LegacyForms in core Grafana
**What is the chore?**:
We have a set of legacy form components, which essentially were deprecated over 3 years ago ( #22862). They all have modern replacement components.

We only [have 11 occurences in core left](https://play.grafana.org/d/VEy5UXjmk/grafana-ci-code-stats?orgId=1&from=1680771811847&to=1696583011847&viewPanel=13), so I though I'd make a list:

`grep -r -o 'LegacyForms;' public/app`:

 - [ ] public/app/core/components/Select/MetricSelect.tsx
 - [x] public/app/plugins/datasource/loki/configuration/QuerySettings.tsx https://github.com/grafana/grafana/pull/76331
 - [x] public/app/plugins/datasource/influxdb/components/editor/config/ConfigEditor.tsx https://github.com/grafana/grafana/pull/76382
 - [x] public/app/plugins/datasource/influxdb/components/editor/config/InfluxFluxConfig.tsx https://github.com/grafana/grafana/pull/76495
 - [ ] public/app/plugins/datasource/influxdb/components/editor/config/InfluxInfluxQLConfig.tsx
 - [x] public/app/plugins/datasource/opentsdb/components/OpenTsdbDetails.tsx https://github.com/grafana/grafana/pull/76165
 - [x] public/app/plugins/datasource/elasticsearch/configuration/DataLink.tsx https://github.com/grafana/grafana/pull/76397
 - [x] public/app/features/admin/ldap/LdapPage.tsx https://github.com/grafana/grafana/pull/76156
 - [x] public/app/features/folders/FolderSettingsPage.tsx https://github.com/grafana/grafana/pull/76313
 - [ ] public/app/angular/angular_wrappers.ts

**Is there anything else we need to know?**:
 In some cases just changing the component might not be enough, as it is located in older code. In these cases, removing occurrences of `gf-form` might also be valuable. If you do, please mention #65513.

**Related issues**
https://github.com/grafana/grafana/issues/66479
https://github.com/grafana/oss-plugin-partnerships/issues/222

### PR DESCRIPTION
**Pull request by Dhoni77**
**Which issue(s) does this PR fix?**:

Related to #76107

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

**Comment by tskarhed** [comment]
Thank you! This file needs some formatting. You could either install our pre-commit hook `make lefthook-install` that solves it for you, or you could run `yarn prettier:write`.

File: public/app/plugins/datasource/opentsdb/components/OpenTsdbDetails.tsx [hunk 1]
```diff
@@ -30,22 +29,16 @@ export const OpenTsdbDetails = (props: Props) => {
 
   return (
     <>
-      <h5>OpenTSDB settings</h5>
-      <div className="gf-form">
-        <InlineFormLabel width={7} htmlFor={`select-version-${idSuffix}`}>
-          Version
-        </InlineFormLabel>
+      <FieldSet label={"OpenTSDB settings"}>
```
Comment (tskarhed) [inline public/app/plugins/datasource/opentsdb/components/OpenTsdbDetails.tsx:32]:
```suggestion
      <FieldSet label="OpenTSDB settings">
```

**Comment by Dhoni77** [comment]
Done the requested changes  @tskarhed @itsmylife 👍

**Comment by tskarhed** [comment]
@itsmylife I am not so sure about using the inline forms. I'll discuss it with some UX people to see what take they have on it.

File: public/app/plugins/datasource/opentsdb/components/OpenTsdbDetails.tsx [hunk 2]
```diff
@@ -30,43 +28,37 @@ export const OpenTsdbDetails = (props: Props) => {
...
+            onChange={onSelectChangeHandler('tsdbResolution', value, onChange)}
+            width={20}
+          />
+        </Field>
+        <Field htmlFor={`lookup-input-${idSuffix}`} label="Lookup limit">
+          <Input
+            id={`lookup-input-${idSuffix}`}
+            type="number"
+            value={value.jsonData.lookupLimit ?? 1000}
+            onChange={onInputChangeHandler('lookupLimit', value, onChange)}
```
Comment (itsmylife) [inline public/app/plugins/datasource/opentsdb/components/OpenTsdbDetails.tsx:58]:
```suggestion
            onChange={onInputChangeHandler('lookupLimit', value, onChange)}
            width={20}
```

**Comment by itsmylife** [comment]
@tskarhed That would be quite helpful. If the `InlineForm` is ok then we can ask this in the epic.

**Comment by tskarhed** [comment]
@itsmylife For the config page, `InlineForm` is a nogo. There is an ongoing migration, which the partner datasources have mainly worked on, but all Grafana owned data sources are on the list.

https://github.com/grafana/grafana/issues/66479
https://github.com/grafana/oss-plugin-partnerships/issues/222

**Review by tskarhed — APPROVED** [review]
It looks good from my side, but I'll let @itsmylife make the final call.

**Comment by catherineymgui** [comment]
I think this is fine.

But can we set a max width to this page and just have all the fields match this?

Suggestion here: https://docs.google.com/document/d/11XMaYHSMSra8AoN2-yVW-5QPbjEXgkAFbTIF6N84cGE/edit#heading=h.5querbxloix5

**Review by itsmylife — APPROVED** [review]

**Pull request merged**